src/spatialdata/_core/operations/_utils.py

```diff
@@ -0,0 +1,120 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from multiscale_spatial_image import MultiscaleSpatialImage
+from spatial_image import SpatialImage
+
+if TYPE_CHECKING:
+    from spatialdata._core.spatialdata import SpatialData
+
+
+def transform_to_data_extent(
+    sdata: SpatialData,
+    coordinate_system: str,
+    target_unit_to_pixels: float | None = None,
+    target_width: float | None = None,
+    target_height: float | None = None,
+    target_depth: float | None = None,
+) -> SpatialData:
+    """
+    Transform the spatial data to match the data extent, keeping the positioning, and making all transformations equal.
```

Comment:
Can this be another operation than scale? If not, I think we should say scale. I think "tranform" is ambiguous. 

Also what is meant by "making all transformations equal"?

Reply:
In `Transform the spatial data to match the data extent, keeping the positioning, and making all transformations equal.`

The `Transform` is referring to the application of the `transform()` operation. The transformation being applied can be anything, not just scale (affine, etc).

The returned object will be such that if we have the point coordinate (x, y), this will correspond to the pixel (y, x). The problem is that in this way we lose the positioning to the original data, so we still need to add scale and translations to make the returned data align with the original data. 

I will add these information to the docstring.

Reply:
I have also added an extra argument, called `maintain_positioning: bool = True`, which if false sets all the transformations to `Identity()`. I explained this argument in the docstring.